### ISSUE TITLE
shim 15.6 for SUSE

### ISSUE THREAD

**Issue by jsegitz** [rhboot/shim-review#263]
Confirm the following are included in your repo, checking each box:

 - [x] completed README.md file with the necessary information
 - [x] shim.efi to be signed
 - [x] public portion of your certificate(s) embedded in shim (the file passed to VENDOR_CERT_FILE)
 - [x] binaries, for which hashes are added to vendor_db ( if you use vendor_db and have hashes allow-listed )
 - [x] any extra patches to shim via your own git tree or as files
 - [x] any extra patches to grub via your own git tree or as files
 - [x] build logs
 - [x] a Dockerfile to reproduce the build of the provided shim EFI binaries

-------------------------------------------------------------------------------
### What is the link to your tag in a repo cloned from rhboot/shim-review?
-------------------------------------------------------------------------------

`https://github.com/jsegitz/shim-review/tree/SUSE-SLES-shim-x86_aarch64-20220712`

-------------------------------------------------------------------------------
### What is the SHA256 hash of your final SHIM binary?
-------------------------------------------------------------------------------
f83baf7da3d799c33806765facd86671ec3426a747f3219d2689958679aa1d32  shim-sles_aarch64.efi
28c14c4e40eb76d226703ac3765dec2226bdbc343d1467893126fc5615a798cb  shim-sles_x86_64.efi

**Comment by jsegitz** [comment]
I'm leaving for a prolonged holiday end of July. It would be great if I could get this done (incl. the signing part by Microsoft) before that

**Comment by parheliamm** [comment]
> I'm leaving for a prolonged holiday end of July. It would be great if I could get this done (incl. the signing part by Microsoft) before that

Hi Johannes:
Thanks for your support in advance.
Chenxi

**Comment by jsegitz** [comment]
Ping. Is there anything blocking this review? Can I help somehow?

**Comment by parheliamm** [comment]
> Ping. Is there anything blocking this review? Can I help somehow?

Hi Johannes:
To be honest, we are waiting for the comments from the shim reviewer at this point.
All tasks have been done and submitted to this repository. I am not sure whether I miss something.
Please give me some suggestions that would be appreciated.

Chenxi

**Comment by msmeissn** [comment]
Cenxi, note this ticket is about the SUSE SLES shim, not the SUSE EULER shim.

**Comment by frozencemetery** [comment]
- Your readme mentions SBAT for fwupdate; are you still using fwupdate for things?  If so, are you aware that fwupdate [has been subsumed into fwupd](https://github.com/rhboot/fwupdate)?
- You appear to have several downstream patches to shim.  Can you speak to your plans for upstreaming them?
- Your README mentions shim 15.4; I assume this should refer to 15.6?
- README.md specifies a command to build (`podman build --build-arg ARCHITECTURE=x86_64 -t sles_shim:15.4`) which I believe is missing the trailing dot.  Please fix.
- Also on the above, is 15.4 correct?  Seems like it should be 15.6?
- Would you be willing to make `podman build .` without arguments do the right thing?  Currently it fails.

**Comment by jsegitz** [comment]
I assume Marcus answered the first two questions.

As for 15.4: There's one place where this was wrong (What patches are being applied and why). The 15.4 in the podman commands etc. refer to our product SLES 15.4. I know that's a bit confusing so I'll change it to 15.6 too, since it doesn't really matter for us

I added the trailing dot, thanks

We can't drop the arguments to podman (apart from the naming for the image) since the Dockerfile is for two architectures that need slightly different commands.By setting ARCHITECTURE this is done. the alternative would be to provide separate Dockerfiles per architecture. If that's better then I can do that.

**Comment by jsegitz** [comment]
This seems stuck. Is there something we can do to help this review?

**Comment by jsegitz** [comment]
ping. Anything we can do to help?

**Comment by jsegitz** [comment]
can someone please reach out to me to discuss when/how this will get reviewed?

**Comment by frozencemetery** [comment]
As you may know, shim reviews are a community effort.  The idea is that distros review each other to ensure quality.  You are at this point intimately aware that the number of reviews is exceeding reviewer capacity.  The best way to expedite your own review is to review submissions from other entities to reduce the pile.

Regarding our earlier conversation about building, I'm still not sure I understand why you need an architecture argument.  If the intent is to make cross-arch building work, it doesn't:

```
$ $ uname -m
x86_64
$ podman build --build-arg ARCHITECTURE=aarch64 -t sles_shim:15.6 .
...
STEP 11/19: RUN zypper -n in /pesign-obs-integration-10.1-13.3.1.$ARCHITECTURE.rpm
Refreshing service 'container-suseconnect-zypp'.
Warning: Skipping service 'container-suseconnect-zypp' because of the above error.
Problem retrieving the repository index file for service 'container-suseconnect-zypp':
[container-suseconnect-zypp|file:/usr/lib/zypp/plugins/services/container-suseconnect-zypp] 
Loading repository data...
Reading installed packages...
'_tmpRPMcache_:pesign-obs-integration=0:10.1-13.3.1' not found in package names. Trying capabilities.
No provider of '_tmpRPMcache_:pesign-obs-integration=0:10.1-13.3.1' found.
Error: error building at STEP "RUN zypper -n in /pesign-obs-integration-10.1-13.3.1.$ARCHITECTURE.rpm": error while running runtime: exit status 104
```

I have access to aa64 hardware and reproduced this there, so I'm not concerned enough to block the review on it, but please fix or remove this for next submission.

**Comment by jsegitz** [comment]
Thank you for the review.

I'm more than happy to help out with the reviews. I tried to get this going in 
https://github.com/rhboot/shim-review/issues/84
but without being able to actually tag submissions as accepted this never went anywhere. Maybe we can create an appointment where bigger stakeholders (e.g. Oracle, SUSE) can discuss how to help out here.

As for cross-arch building: This is there because of this and it should work. I'll look into the issue, didn't happen for me. I think this is a probelm with our repos that didn't happen when I worked through this
